### PR TITLE
Adding replication connection option

### PR DESCRIPTION
**Pull request by David-Engel**
The replication property simply tells the database that a connection is for replication and affects when the "NOT FOR REPLICATION" setting on triggers and constraints takes affect.

This is the feature requested in #1058.

### CODE REVIEW

File: src/test/java/com/microsoft/sqlserver/jdbc/connection/ReplicationTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+@RunWith(JUnitPlatform.class)
+@Tag(Constants.xAzureSQLDW)
+public class ReplicationTest extends AbstractTest {
+
+    @Test
+    public void testReplication() throws SQLException {
+        String tableName = RandomUtil.getIdentifier("repl");
+        String triggerName = RandomUtil.getIdentifier("trig");
+        String escapedTableName = DBTable.escapeIdentifier(tableName);
+        String escapedTriggerName = DBTable.escapeIdentifier(triggerName);
```
Comment (lilgreenbird) [inline src/test/java/com/microsoft/sqlserver/jdbc/connection/ReplicationTest.java:40]: don't use DBTable, that is old framework code that we are moving away from

File: src/test/java/com/microsoft/sqlserver/jdbc/connection/ReplicationTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        SQLServerDataSource ds = new SQLServerDataSource();
+        ds.setURL(connectionString);
+        ds.setReplication(false);
+
+        String sqlDropTable = "IF OBJECT_ID('" + TestUtils.escapeSingleQuotes(escapedTableName) + "', 'U') IS NOT NULL "
```
Comment (lilgreenbird) [inline src/test/java/com/microsoft/sqlserver/jdbc/connection/ReplicationTest.java:45]: use TestUtils.dropTableIfExists?

File: src/test/java/com/microsoft/sqlserver/jdbc/connection/ReplicationTest.java (hunk 1 above)
Comment (David-Engel) [reply src/test/java/com/microsoft/sqlserver/jdbc/connection/ReplicationTest.java:40]: Is there a current utility method that replaces it?

Comment (lilgreenbird) [reply src/test/java/com/microsoft/sqlserver/jdbc/connection/ReplicationTest.java]: oh just use AbstractSQLGenerator.escapeIdentifier(tableName). As an example, see SQLServerConnectTest.java

**Review by lilgreenbird — APPROVED** [review]

**Pull request merged**